pkg/util/provider/machinecontroller/machine_safety.go

```diff
@@ -265,15 +266,18 @@ func (c *controller) checkMachineClass(ctx context.Context, machineClass *v1alph
 		machine, err := c.machineLister.Machines(c.namespace).Get(machineName)
 
 		if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
It would be much easier to understand without two negations, for example

```go
if err != nil {
    if apierros.IsNotFound(err) {
...
  }
}
```

Reply:
Yes @majst01 , thats what I have addressed in the latest commit. Thanks for commenting.
Please let me know your thoughts regarding the new flow.